PWGJE/Tasks/emccellmonitor.cxx

```diff
@@ -0,0 +1,215 @@
+        mCellTimeMain->Fill(cell.time(), cell.cellNumber());
+
+      // Get Cell index in eta-phi of sm
+      auto [supermodule, module, phiInModule, etaInModule] = mGeometry->GetCellIndex(cell.cellNumber());
+      auto [row, col] = mGeometry->GetCellPhiEtaIndexInSModule(supermodule, module, phiInModule, etaInModule);
+
+      auto smhists = mSupermoduleHistMapper.find(supermodule);
+      if (smhists != mSupermoduleHistMapper.end()) {
+        smhists->second.mCellCount->Fill(col, row);
+        smhists->second.mCellAmp->Fill(col, row, cell.amplitude());
+        smhists->second.mCellAmpTime->Fill(cell.time(), cell.amplitude());
+      }
+    }
+  }
+
+  /// \brief Check if a cell is masked
+  /// \param towerID ID of of the tower to be checked
+  /// \return true if the tower is masked, false if the tower is considered good
+  ///
+  /// Requires presence of the bad channel map from the CCDB
+  bool isCellMasked(int towerID)
+  {
+    bool masked = false;
+    if (mBadChannels) {
+      auto maskStatus = mBadChannels->getChannelStatus(towerID);
+      masked = (maskStatus != o2::emcal::BadChannelMap::MaskType_t::GOOD_CELL);
+    }
+    return masked;
+  }
+
+  /// \brief Create binning for cell amplitude axis (variable bin size)
+  /// \return vector with bin limits
+  std::vector<double> makeAmplitudeBinning() const
+  {
+    auto fillBinLimits = [](std::vector<double>& binlimits, double max, double binwidth) {
+      auto current = *binlimits.rbegin();
+      while (current < max) {
+        current += binwidth;
+        binlimits.emplace_back(current);
+      }
+    };
+    std::vector<double> result = {0.};
+    fillBinLimits(result, 2., 0.1);
+    fillBinLimits(result, 5., 0.2);
+    fillBinLimits(result, 10., 0.5);
+    fillBinLimits(result, 20., 1.);
+    fillBinLimits(result, 50., 2.);
+    fillBinLimits(result, 100., 5.);
+    fillBinLimits(result, 200., 10.);
+    return result;
+  }
+
+  /// \brief Helper function creating a vector with bin limits for a linear binning (const bin width)
+  /// \param nbins Number of bins
+  /// \param xmin Min. of the range represented by the binning
+  /// \param xmax Max. of the range represented by the binning
+  /// \return vector with bin limits
+  std::vector<double> makeLinearBinning(int nbins, double xmin, double xmax) const
+  {
+    double binwidth = (xmax - xmin) / nbins;
+
+    std::vector<double> result;
+    double currentmin = xmin;
+    result.push_back(currentmin);
+    while ((currentmin < xmax) && (std::abs(currentmin - xmax) > DBL_EPSILON)) {
+      currentmin += binwidth;
+      result.push_back(currentmin);
+    }
+    return result;
+  }
+};
+} // namespace o2::emcal
+
+o2::framework::WorkflowSpec defineDataProcessing(o2::framework::ConfigContext const& cfgc)
+{
+  return o2::framework::WorkflowSpec{
+    o2::framework::adaptAnalysisTask<o2::emcal::CellMonitor>(cfgc, o2::framework::TaskName{"emcal-cellmonitor-task"})};
```

Comment:
can you drop TaskName

Reply:
Seems this is compatible only with tasks in the global namespace.

Reply:
I see. Any need for the name space? The struct only exists in this executable and is never included anymore else.

Reply:
Yes, indeed. This is the case for the O2Physics tasks. Maybe you can remove the emcal namespace here, such that we can use the canonical task name definition. The other items I leave to you to decide.

Reply:
Will do, future components of an analysis framework library should however go to the namespace o2::emcal (AOD content to o2::aod) for safety reasons.